tests/test_httpstatus.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8
+
+import ddt
+
+import falcon.testing as testing
+import falcon
+from falcon.http_status import HTTPStatus
+
+
+def before_hook(req, resp, params):
+    raise HTTPStatus(falcon.HTTP_200,
+                     headers={"X-Failed": "False"},
+                     body="Pass")
+
+
+def after_hook(req, resp, resource):
+    resp.status = falcon.HTTP_200
+    resp.set_header("X-Failed", "False")
+    resp.body = "Pass"
+
+
+class TestStatusResource:
+
+    @falcon.before(before_hook)
+    def on_get(self, req, resp):
+        resp.status = falcon.HTTP_500
+        resp.set_header("X-Failed", "True")
+        resp.body = "Fail"
+
+    def on_post(self, req, resp):
+        resp.status = falcon.HTTP_500
+        resp.set_header("X-Failed", "True")
+        resp.body = "Fail"
+
+        raise HTTPStatus(falcon.HTTP_200,
+                         headers={"X-Failed": "False"},
+                         body="Pass")
+
+    @falcon.after(after_hook)
+    def on_put(self, req, resp):
+        raise HTTPStatus(falcon.HTTP_500,
+                         headers={"X-Failed": "True"},
+                         body="Fail")
+
+
+@ddt.ddt
+class TestHTTPError(testing.TestBase):
+    def before(self):
+        self.resource = TestStatusResource()
+        self.api.add_route('/status', self.resource)
+
+    def test_raise_status_in_before_hook(self):
+        body = self.simulate_request('/status', method='GET', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_in_responder(self):
+        body = self.simulate_request('/status', method='POST', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_runs_after_hooks(self):
+        body = self.simulate_request('/status', method='POST', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
```

Comment:
I'd like to see one more test where the hook does not override the middleware, just to prove that the body set in HTTPStatus is returned as expected.